next/components/molecules/SearchPageNew/useQueryBySearchOption.ts

```diff
@@ -22,28 +23,43 @@ import {
   pagesFetcherUseQuery,
   PagesFilters,
 } from '@backend/meili/fetchers/pagesFetcher'
+import {
+  getRegulationsQueryKey,
+  RegulationFilters,
+  regulationsFetcher,
+} from '@backend/meili/fetchers/regulationsFetcher'
 import { PageMeili } from '@backend/meili/types'
 import {
   getMsGraphSearchQueryKey,
   msGraphSearchFetcher,
 } from '@backend/ms-graph/fetchers/msGraphSearch.fetcher'
 import { SearchOption } from '@components/pages/searchPageContentNew'
 import { useQuery } from '@tanstack/react-query'
+import { isDefined } from '@utils/isDefined'
 import { formatDate } from '@utils/local-date'
-import { useLocale } from 'next-intl'
+import { findIconByPageColor } from '@utils/pageIcons'
+import DocumentListCategorysMap, { findIconByRegulationCategory } from '@utils/regulationIcons'
+import { useLocale, useTranslations } from 'next-intl'
+import { FunctionComponent, SVGProps } from 'react'
 
-export type SearchFilters = PagesFilters | BlogPostsFilters | InbaArticlesFilters
+export type SearchFilters =
+  | PagesFilters
+  | BlogPostsFilters
+  | InbaArticlesFilters
+  | RegulationFilters
 
 export type SearchResult = {
   title: string | null | undefined
   linkHref?: string | null | undefined
   metadata?: (string | null | undefined)[]
   coverImageSrc?: string | null | undefined
-  pageColor?: Enum_Page_Pagecolor | Enum_Pagecategory_Color
+  pageColor?: Enum_Page_Pagecolor | Enum_Pagecategory_Color | null
   customIconName?: string
+  customIcon?: React.ReactNode | FunctionComponent<SVGProps<SVGSVGElement>> | undefined
```

Comment:
`FuncionalComponent` is probably subset of `ReactNode`.
`ReactNode` also contains `null`.
I does not make sense to use `?` and `| undefined` here probably. If you need to tell in parent component, that there is no icon, use null.